### PR TITLE
Subentities

### PR DESCRIPTION
**Pull request by chrisforbes**
This is about half of what we need for subentities. They work now if you wire them up manually in code. Next step is to allow children to be specified in entity specs

### CODE REVIEW

**Comment by chrisforbes** [comment]
Now we can specify children in entity specs. Remaining piece is to be able to specify a matrix in spec.

**Comment by chrisforbes** [comment]
This should be good to go now.

File: src/component/component_manager.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ struct entity_data {
     }
 
     // disable copy
-    entity_data(const entity_data &) = default;
+    entity_data(const entity_data &) = delete;
```
Comment (RobotCaleb) [inline src/component/component_manager.h]: lol whoops

File: src/component/component_system_manager.h [hunk 2]
```diff
@@ -54,4 +54,35 @@ void
...
+template<>
+static inline
+int load_value_from_config<int>(config_setting_t const *s, char const *key) {
+    auto m = config_setting_get_member(s, key);
+    return config_setting_get_int(m);
+}
+
+template<>
+static inline
+placement load_value_from_config<placement>(config_setting_t const *s, char const *key) {
```
Comment (RobotCaleb) [inline src/component/component_system_manager.h]: should this go in the enum stuff?

File: src/component/component_system_manager.h [hunk 3]
```diff
@@ -54,4 +54,35 @@ void
 draw_doors(frame_data *frame);
 
 void
-build_absolute_transforms();
\ No newline at end of file
+build_absolute_transforms();
+
+template<typename T>
+T load_value_from_config(config_setting_t const *s, char const *key) = delete;
```
Comment (RobotCaleb) [inline src/component/component_system_manager.h]: Cool!

**Comment by RobotCaleb** [comment]
these got merged, just github doesn't think so.

**Pull request closed without merging**